docs/semgrep-ci/manual-ci-setup.md

```diff
@@ -0,0 +1,415 @@
+:::
+
+#### GitLab CI/CD code snippet
+
+```yaml
+semgrep:
+
+  # A Docker image with Semgrep installed.
+  image: returntocorp/semgrep
+
+  # Run the "semgrep ci" command on the command line of the docker image.
+  script: semgrep ci
+
+  rules:
+    # Scan changed files in MRs, only report new findings (existing findings ignored)
+    - if: $CI_MERGE_REQUEST_IID
+    # Scan all files on the default branch, report any findings.
+    - if: $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
+
+  variables:
+
+    # Define rules through the SEMGREP_RULES environment variable. 
+    SEMGREP_RULES: p/default
+
+    # Change job timeout (value in seconds; default is 1800 seconds. Set to 0 to disable)
+    #   SEMGREP_TIMEOUT: 300
+```
+
+#### GitLab CI/CD code snipppet with GitLab SAST
+
+```yaml
+semgrep:
+
+  # A Docker image with Semgrep installed.
+  image: returntocorp/semgrep
+
+  rules:
+
+    # Scan changed files in MRs, only report new findings (existing findings ignored)
+    - if: $CI_MERGE_REQUEST_IID
+  
+    # Scan all files on the default branch, report any findings.
+    - if: $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
+
+  variables:
+
+    # Define rules through the SEMGREP_RULES environment variable. 
+    SEMGREP_RULES: p/default # See more at semgrep.dev/explore.
+
+    # Change job timeout (value in seconds; default is 1800 seconds. Set to 0 to disable)
+    # SEMGREP_TIMEOUT: 300
+  
+    # Upload findings to GitLab SAST Dashboard
+    SEMGREP_GITLAB_JSON: "1"
+    script: semgrep ci --gitlab-sast > gl-sast-report.json || true
+    artifacts:
+      reports:
+        sast: gl-sast-report.json
+```
+
+### Jenkins
+
+To add Semgrep into your Jenkins pipeline, do the following steps. Your UI may vary depending on your Jenkins installation. These steps use a Classic UI Jenkins interface.
+
+1. Edit or create your `Jenkinsfile` configuration file to add a Semgrep `step` as part of your Pipeline. Refer to the [Jenkins CI code snippet](#jenkins-ci-code-snippet). You can edit your `Jenkinsfile` from Jenkins's interface.
+2. Optional: You can also review [`Jenkinsfile` usage](https://www.jenkins.io/doc/book/pipeline/jenkinsfile/) to help with editing the configuration file.
+3. If you are using Jenkins Classic UI, [save your `Jenkinsfile`](https://www.jenkins.io/doc/book/pipeline/getting-started/#through-the-classic-ui) within the Jenkins interface. Otherwise, commit the configuration file into the root folder within the target repository. The Semgrep job starts automatically upon detecting the `Jenkinsfile` commit in the previous step. 
+4. Optional: Re-run the stage from [Jenkins's interface](https://www.jenkins.io/doc/book/pipeline/running-pipelines/#restarting-from-the-classic-ui).
+6. Customize the job's behavior, such as its rules to scan and files to ignore.
+
+#### Jenkins CI code snippet
+
+This code snippet uses Jenkins declarative syntax.
+
+```javascript
```

Comment:
Jenkinsfile is resembles javascript (not yaml) so I used that for syntax highlighting. Hope this is okay. Opinions welcome.

Reply:
(I hope my previous responses answer this question as to why the "commented out" vars aren't here.)